internal/graph/check_fast_path.go

```diff
@@ -629,14 +632,18 @@ func fanInIteratorChannels(ctx context.Context, chans []chan *iterator.Msg) chan
 	pool := concurrency.NewPool(ctx, limit)
 
 	for _, c := range chans {
+		if c == nil {
+			continue
+		}
 		pool.Go(func(ctx context.Context) error {
 			for v := range c {
-				if !concurrency.TrySendThroughChannel(ctx, v, out) {
+				if v != nil && !concurrency.TrySendThroughChannel(ctx, v, out) {
 					if v.Iter != nil {
 						v.Iter.Stop()
 					}
 				}
 			}
+
 			return nil
 		})
 	}
```

Comment:
There's currently no way I can see to handle this goroutine panicking, since the only way for the above `pool.Go` to panic would be if the `!concurrency.TrySendThroughChannel(...)` call is true which would require the ctx to be cancelled.